src/codegen/Translator.cpp

```diff
@@ -1931,5 +1931,40 @@ code::ForStmt* Translator::createForeachLoop(std::string loop_var, std::string l
   return createForeachLoop(loop_var, new code::Identifier(loop_n), body, isVarDefined, isLess);
 }
 
+code::Stmt* Translator::createMultiArgForeachLoop(std::vector<std::string>& loop_vars, std::vector<int> argDim, code::Stmt* body) {
+  code::Stmt* retStmt = body;
+  for (int k = (int)loop_vars.size() - 1; k >= 0; --k) {
+    retStmt = createForeachLoop(loop_vars[k],
+      new code::IntegerLiteral(argDim[k]), retStmt, false, true);
+  }
+  return retStmt;
+}
+
+code::Expr* Translator::get_var_name_with_args(std::string name, std::vector<std::string>& args) {
+  code::Expr* var = new code::Identifier(name);
+  for (size_t i = 0; i < args.size(); ++i) {
+    var = new code::ArraySubscriptExpr(var, new code::Identifier(args[i]));
+  }
+  return var;
+}
+
+code::Expr* Translator::printer(std::string qr_string, std::vector<std::string>& vecstr_names,
```

Comment:
@giulio-zhou can you create a more informative method name? :)

Also I think for this function better to add some comments above it.